examples/pong_tutorial_04/systems/move_balls.rs

```diff
@@ -1,26 +1,21 @@
 use amethyst::{
     core::{timing::Time, transform::Transform},
-    derive::SystemDesc,
-    ecs::prelude::{Join, Read, ReadStorage, System, SystemData, WriteStorage},
+    ecs::{Runnable, SystemBuilder},
+    prelude::*,
 };
 
 use crate::pong::Ball;
 
-#[derive(SystemDesc)]
-pub struct MoveBallsSystem;
-
-impl<'s> System<'s> for MoveBallsSystem {
-    type SystemData = (
-        ReadStorage<'s, Ball>,
-        WriteStorage<'s, Transform>,
-        Read<'s, Time>,
-    );
-
-    fn run(&mut self, (balls, mut locals, time): Self::SystemData) {
-        // Move every ball according to its speed, and the time passed.
-        for (ball, local) in (&balls, &mut locals).join() {
-            local.prepend_translation_x(ball.velocity[0] * time.delta_seconds());
-            local.prepend_translation_y(ball.velocity[1] * time.delta_seconds());
-        }
-    }
+pub fn build() -> impl Runnable {
+    SystemBuilder::new("MoveBallsSystem")
+        .with_query(<(&Ball, &mut Transform)>::query())
+        .read_resource::<Time>()
+        .read_component::<Ball>()
+        .write_component::<Transform>()
+        .build(move |_commands, world, time, query_balls| {
```

Comment:
Once again, I wonder if using #[system] macro would help as described in 
https://amethyst.rs/posts/legion-ecs-v0.3